simulator/scheduler/scheduler.go

```
@@ -67,124 +64,65 @@ func NewSchedulerService(client clientset.Interface, restclientCfg *restclient.C
 	return &Service{clientset: client, restclientCfg: restclientCfg, initialSchedulerCfg: initCfg, sharedStore: sharedStore, simulatorPort: simulatorPort}
 }
 
-func (s *Service) RestartScheduler(cfg *configv1.KubeSchedulerConfiguration) error {
-	if s.disabled {
-		return xerrors.Errorf("an external scheduler is enabled: %w", ErrServiceDisabled)
+func restartContainer(ctx context.Context, cli *client.Client, cfg *configv1.KubeSchedulerConfiguration) error {
+	containers, err := cli.ContainerList(ctx, container.ListOptions{})
+	if err != nil {
+		return xerrors.Errorf("failed to get container list: %w", err)
 	}
-
-	s.ShutdownScheduler()
-
-	oldSchedulerCfg := s.currentSchedulerCfg
-	if err := s.StartScheduler(cfg); err != nil {
-		klog.Infof("failed to start scheduler: %v. restarting with old configuration", err)
-		if err2 := s.StartScheduler(oldSchedulerCfg); err2 != nil {
-			klog.Warningf("failed to start scheduler with old configuration: %v", err2)
-			return xerrors.Errorf("start scheduler: %w, restart scheduler with old configuration: %w", err, err2)
+	for _, c := range containers {
```

We should return an error when we cannot find the scheduler.